addons/hr_holidays/static/src/components/hr_presence_status_private/hr_presence_status_private.js

```diff
@@ -0,0 +1,38 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { HrPresenceStatus } from "@hr/components/hr_presence_status/hr_presence_status";
+import { _t } from "@web/core/l10n/translation";
+class HrPresenceStatusPrivate extends HrPresenceStatus {
+    get icon() {
+        if (this.value.startsWith("presence_holiday")) {
+            return "fa-plane";
+        }
+        return super.icon;
+    }
+
+    get color() {
+        if (this.value.startsWith("presence_holiday")) {
+            return `text-${this.value === "presence_holiday_present" ? "success" : "warning"}`;
+        }
+        return super.color;
+    }
+
+    get label() {
+        return this.props.record.data["current_leave_name"] != ""
+            ? this.props.record.data["current_leave_name"]
+            : super.label;
+    }
+}
+
+export const hrPresenceStatusPrivate = {
+    component: HrPresenceStatusPrivate,
+    displayName: _t("HR Presence Status For Private Employee"),
+    extractProps({ viewType }, dynamicInfo) {
+        return {
+            tag: viewType === "kanban" ? "span" : "small",
+        };
+    },
+};
```

Comment:
We also have to add `current_leave_id ` field in the field dependencies to be sure we will fetch it. 🙂 

```suggestion
export const hrPresenceStatusPrivate = {
    ...hrPresenceStatus,
    component: HrPresenceStatusPrivate,
    displayName: _t("HR Presence Status For Private Employee"),
    extractProps({ viewType }, dynamicInfo) {
        return {
            tag: viewType === "kanban" ? "span" : "small",
        };
    },
    fieldDependencies: [
        ...(hrPresenceStatus.fieldDependencies || []),
        { name: "current_leave_id", type: "many2one" },
    ],
};
```